/*
 * This file was automatically generated by EvoSuite
 * Thu Apr 21 16:50:15 GMT 2022
 */

package org.apache.manifoldcf.core.database;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.manifoldcf.core.database.RSet;
import org.apache.manifoldcf.core.interfaces.IResultRow;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class RSet_ESTest extends RSet_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      RSet rSet0 = new RSet();
      rSet0.addRow((IResultRow) null);
      int int0 = rSet0.getRowCount();
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      RSet rSet0 = new RSet();
      rSet0.addRow((IResultRow) null);
      rSet0.addRow((IResultRow) null);
      rSet0.addRow((IResultRow) null);
      IResultRow iResultRow0 = rSet0.getRow(2);
      assertNull(iResultRow0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      RSet rSet0 = new RSet();
      rSet0.rows = null;
      // Undeclared exception!
      try { 
        rSet0.getRows();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.manifoldcf.core.database.RSet", e);
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      RSet rSet0 = new RSet();
      rSet0.rows = null;
      // Undeclared exception!
      try { 
        rSet0.getRowCount();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.manifoldcf.core.database.RSet", e);
      }
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      RSet rSet0 = new RSet();
      rSet0.rows = null;
      // Undeclared exception!
      try { 
        rSet0.getRow(4);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.manifoldcf.core.database.RSet", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      RSet rSet0 = new RSet();
      // Undeclared exception!
      try { 
        rSet0.getRow((-361));
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      RSet rSet0 = new RSet();
      rSet0.rows = null;
      // Undeclared exception!
      try { 
        rSet0.addRow((IResultRow) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.manifoldcf.core.database.RSet", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      RSet rSet0 = new RSet();
      IResultRow[] iResultRowArray0 = rSet0.getRows();
      assertEquals(0, iResultRowArray0.length);
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      RSet rSet0 = new RSet();
      int int0 = rSet0.getRowCount();
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      RSet rSet0 = new RSet();
      // Undeclared exception!
      try { 
        rSet0.getRow(2);
        fail("Expecting exception: IndexOutOfBoundsException");
      
      } catch(IndexOutOfBoundsException e) {
         //
         // Index: 2, Size: 0
         //
         verifyException("java.util.ArrayList", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      RSet rSet0 = new RSet();
      rSet0.addRow((IResultRow) null);
      IResultRow[] iResultRowArray0 = rSet0.getRows();
      assertEquals(1, iResultRowArray0.length);
  }
}
